crates/ibc/src/core/ics04_channel/handler/write_acknowledgement.rs

```diff
@@ -56,10 +56,6 @@ pub fn process<Ctx: ChannelReader>(
         Err(e) => return Err(e),
```

Comment:
Note that this is legacy code and is never called from `Validation/ExecutionContext`. We no longer treat `PacketAcknowledgementNotFound` specially in the [new implementation](https://github.com/cosmos/ibc-rs/blob/dc84bf2e8359f4bfcb78e3a8cbb9cad3a1c47002/crates/ibc/src/core/ics04_channel/handler/recv_packet.rs#L153-L161).

> I think get_packet_acknowledgement(...) should return Result<Option<AcknowledgementCommitment>, PacketError> type, so under normal circumstance, the None variant would show a valid absence of ACK

What's a "valid" absence of an ACK (as opposed to an "invalid one")? The (undocumented) contract of `get_packet_acknowledgement()` is "return an ack commitment, or tell me why you were not able to do it". The acknowledgement not being present is just another reason why the host would not be able to return it, and is an error in this context. IMO `Result<Option<AcknowledgementCommitment>, PacketError>` is harder to read, and doesn't provide us any gains that I can see. Note also that this would make the new implementation more complex, as we'd not only need to check whether or not there was an error, but also if the return was `None` in the success case.